rename.g

```diff
@@ -0,0 +1,324 @@
+## The renamings_to_execute.csv file specifies how to rename.
+## The renamings_history.csv file stores the history of renamings
+## and is used for the generation of the renaming table in the documentation.
+## --------------------------------------------------------------------------
+## INSTRUCTIONS:
+## 
+## Run it with GAP from within the package directory.
+## There are two possible applications for running this file.
+## --------------------------------------------------------------------------
+## APPLICATIONS:
+## 
+## The first one adresses the developers of the recog package.
+## They can use the renamings_to_execute.csv file in order to
+## discuss new names for functions and afterwards to apply the name changes
+## in the recog package by running this file as instructed.
+## Below an example of this application is included.
+## 
+## The second one adresses people who use the recog package as a dependency.
+## They can run this file with an empty renamings_to_execute.csv file
+## to apply all name changes that are stored in the renamings_history.csv 
+## file to their own projects.
+## --------------------------------------------------------------------------
+## DETAILS:
+## 
+## This file reads the renamings_to_execute.csv file that specifies how to
+## rename. If it is empty, this file attempts to do the renamings specified
+## in the renamings_history.csv file. 
+## The further explanations apply if the file is not empty.
+## 
+## First this file checks if the renamings_to_execute.csv file is correct.
+## The renamings_to_execute.csv file has the following form:
+##     - First column 'OLD NAME' contains the old name.
+##     - Second column 'NEW NAME' contains the new name that is used now.
+##     - Third column 'TYPE' contains the type used for Ref in GAPDoc.
+## Then further checks are done and on failure Errors/Warnings are printed.
+## Primarily they are done to ensure that the renamings_history.csv file 
+## always has duplicate-free columns after merging. 
+## After successful checks all matches on word boundaries of the old name get
+## replaced with the new name. The replacements are executed in all files with 
+## extensions .g/.gd/.gi/.tst/.xml found in the directory tree rooted at '.' 
+## whereas all files in the .git folder are ignored.
+## Afterwards the rows from the renamings_to_execute.csv file are added to 
+## the renamings_history.csv file. Then the renaming table for the 
+## documentation is generated from the renamings_history.csv file whereby
+## renamings of the form 'X_1 -> X_2 -> ... -> X_n' are printed 
+## in the documentation as 'X_1 -> X_n'.
+## --------------------------------------------------------------------------
+## EXAMPLE:
+## 
+## renamings_to_execute.csv :
+## OLD NAME,NEW NAME,TYPE
+## undesciptive,descriptive,Attr
+## 
+## First add and commit the renamings_to_execute.csv file. -- Commit 1
+## Then discuss on github if the suggested names are suitable.
```

Comment:
In principle people can also directly create what you call `Commit 1` and `Commit 2` and push that to github. So I'd remove this line and change the next line to
```## Then run `gap rename.g` from within the package directory.```